### PR TITLE
Add `decision_function` to return raw logits

### PR DESCRIPTION
**Pull request by singjc**
This PR adds a `decision_function` (same naming convention as scikit-learn), providing users with direct access to raw decision scores from the gradient boosting model.

### CODE REVIEW

**Comment by singjc** [comment]
Hi @dingelish , any updates on this being merged?
